catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/FileSystemPersistenceProvider.java

```diff
@@ -211,6 +212,20 @@ public void clear() {
     }
   }
 
+  public void storeJson(String key, Object value, Class<?> aClass) {
+    String jsonToWrite = new Gson().toJson(value, aClass);
+    store(key, jsonToWrite);
```

Comment:
❗️Unless I'm mistaken, this will actually use Java object serialization and serialize the `jsonToWrite` String instead of writing the JSON to a text file like we want.

Reply:
Yeah you're correct and that was the hack. 😄
What would you suggest to make this cleaner? I saw your post [above](https://github.com/codice/ddf/pull/4223#discussion_r253621334). If we go for that route then should this code exist outside of the CDM package?